src/python/pants/backend/rust/lint/rustfmt/rules.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pants.backend.rust.lint.rustfmt import skip_field
+from pants.backend.rust.lint.rustfmt.skip_field import SkipRustfmtField
+from pants.backend.rust.lint.rustfmt.subsystem import RustfmtSubsystem
+from pants.backend.rust.target_types import RustPackageSourcesField
+from pants.backend.rust.util_rules.toolchains import RustToolchainProcess
+from pants.core.goals.fmt import FmtResult, FmtTargetsRequest
+from pants.core.util_rules.partitions import PartitionerType
+from pants.engine.internals.selectors import Get
+from pants.engine.process import ProcessResult
+from pants.engine.rules import collect_rules, rule
+from pants.engine.target import FieldSet, Target
+from pants.util.logging import LogLevel
+from pants.util.strutil import pluralize
+
+
+@dataclass(frozen=True)
+class RustfmtFieldSet(FieldSet):
+    required_fields = (RustPackageSourcesField,)
+
+    sources: RustPackageSourcesField
+
+    @classmethod
+    def opt_out(cls, tgt: Target) -> bool:
+        return tgt.get(SkipRustfmtField).value
+
+
+class RustfmtRequest(FmtTargetsRequest):
+    field_set_type = RustfmtFieldSet
+    tool_subsystem = RustfmtSubsystem
+    partitioner_type = PartitionerType.DEFAULT_SINGLE_PARTITION
+
+
+@rule(desc="Format with rustfmt")
+async def rustfmt_fmt(request: RustfmtRequest.Batch, rustfmt: RustfmtSubsystem) -> FmtResult:
+    print("rustfmt_fmt")
+    args = (
+        *rustfmt.args,
+        # Filter out non-.rs files, e.g. cargo.toml, from the file list.
+        *(f for f in request.snapshot.files if f.endswith(".rs")),
```

Comment:
This shouldn't be an issue anymore, with the new sources

Reply:
Aint the value I provided just a default value? Meaning cant anyone overwrite sources within the target anyways?

Reply:
Yes, but at that point they're explicitly opting into marking things as rust sources. We should do as they say

Reply:
I can accept it, if for example you want to insert rust code into a separate filetype as compared to `.rs`. 

May I suggest that such rule is enforced across the project? This is not uniquely done as part of this PR, and it should happen that we try and align behaviour across languages as much as possible.

Reply:
I'm pretty sure other tools already just accept the sources as-is. If they don't it's a bug.

Reply:
To be clear, I wouldn't mark this as resolved. We should just pass through what the user provides into the process

Reply:
Ugh, sorry the GitHub app misled me into thinking the code hasn't changed.